project/src/components/favorite-list/favotite-list.jsx

```diff
@@ -0,0 +1,87 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { getStarRaiting } from '../../common';
+
+function FavoriteList(props) {
+  const { offers } = props;
+  const citys = [];
+
+  for (let i = 0; i < offers.length; i++) {
+    if (!citys.includes(offers[i].city.name) && offers[i].isFavorite) {
+      citys.push(offers[i].city.name);
+    }
+  }
```

Comment:
Отдавай предпочтение методам массива
```suggestion
  const citys = offers.reduce((acc, offer) => {
    const hasCity = acc.includes(offer.city.name) && offer. isFavorite;
    
    return hasCity ? acc : acc.concat(offer.city.name);
  }, [])
  }
```

а ещё лучше вынести это в функцию, и вообще вынести этот расчет с компонента, а оставь тут только вызов функции